nucypher/blockchain/eth/token.py

```diff
@@ -641,11 +641,8 @@ def __tracking_consistency_check(self) -> bool:
             return True  # OK!
         if txs_in_mempool > len(self.__pending):  # We're missing some pending TXs
             return False
-
-        # TODO: Not sure what to do in this case, but let's do this for the moment
-        #       Note that the block my have changed since the previous query
-        # elif txs_in_mempool < len(self.__pending):  # Our tracking is somehow outdated
-        #     return False
+        else:  # TODO: What to do when txs_in_mempool < len(self.__pending)? What does this imply?
+            return True
```

Comment:
If I understand the logic correctly, it could happen if somehow some transactions were canceled externally. 

Feels like it would be better to at least return `False`, or maybe even raise an exception/write something in log in this case. 

An interesting side effect here is that if `txs_in_mempool == len(self.__pending) - 1`, and we return `False`, `__track_pending_commitments()` adds an element in `self.__pending`, and the subsequent calls to `__tracking_consistency_check()` return `True` again. Maybe it's worth using a different variable to store that state.

Reply:
> If I understand the logic correctly, it could happen if somehow some transactions were canceled externally.

Yes - this might be from a transaction unassociated with nucypher, broadcasted before startup.

Reply:
This block also may execute for a pending period commitment transaction, right after a node manually (re)starts.